AgentDiscoveries-Backend/src/main/java/org/softwire/training/api/routes/v1/AgentsRoutes.java

```diff
@@ -34,13 +35,27 @@ public Agent createAgent(Request req, Response res) throws FailedRequestExceptio
     }
 
     public Agent readAgent(Request req, Response res, int id) throws FailedRequestException {
+        int userId = req.attribute("user_id");
+        if(id!=userId){
+            throw new FailedRequestException(ErrorCode.INVALID_CREDENTIALS, "User not authorised");
+        }

```

Comment:
This (and most of the routes below) have the same issue with permissions etc as did the admin one; (Wrong error code) when merged with that I'll make sure the admin permission set up is implemented!

Reply:
OK thanks!